### DISCUSSION TITLE
ArmPkg/Library/CompilerIntrinsicsLib/CompilerIntrinsicsLib.inf is not found in packages path

### DISCUSSION THREAD

**Discussion by r1chard-lyu** [tianocore/edk2#6503] [Q&A] [answered]
I tried to build the SCT and followed the instructions provided [here](https://github.com/tianocore/edk2-test/blob/master/uefi-sct/HowToBuild/HowToBuildSct.txt) on Linux, but encountered the following error:

build.py...
/home/Workspace/sct_workspace/SctPkg/UEFI/UEFI_SCT.dsc(179): error 000E: File/directory not found in workspace
    ArmPkg/Library/CompilerIntrinsicsLib/CompilerIntrinsicsLib.inf is not found in packages path:
    /home/Workspace/sct_workspace/edk2
    /home/Workspace/sct_workspace/SctPkg

It seems CompilerIntrinsicsLib.inf is missing from the edk2. Does anyone have any ideas on how to resolve this?

**Comment by anandhkrishna** [comment]
@r1chard-lyu 
it seems CompilerIntrinsicsLib.inf moved from ArmPkg to MdePkg from this commit [MdePkg: Move CompilerIntrinsicsLib from ArmPkg](https://github.com/tianocore/edk2/commit/8f74b95a21cf106fa4eb4932e22b404c57297ba2) .
i would like you to try modify the CompilerIntrinsicsLib.inf path in UEFI_SCT.dsc

**Comment by anandhkrishna** [comment]
@r1chard-lyu try this changes [https://github.com/tianocore/edk2-test/pull/105](https://github.com/tianocore/edk2-test/pull/105) or you can even try to use tagged version

**Comment by r1chard-lyu** [comment]
@anandhkrishna Thank you for providing the information. After applying the patch for the change that moved the library from ArmPkg to MdePkg, I encountered another error, which appears to be a different issue, as shown below.

```text
make: *** [GNUmakefile:291: /home/richard/Workspace/sct_workspace/Build/UefiSct/RELEASE_GCC49/AARCH64/MdePkg/Library/StackCheckLibNull/StackCheckLibNull/OUTPUT/StackCheckLibNullGcc.obj] Error 127


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/richard/Workspace/sct_workspace/Build/UefiSct/RELEASE_GCC49/AARCH64/MdePkg/Library/StackCheckLibNull/StackCheckLibNull]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/richard/Workspace/sct_workspace/Build/UefiSct/RELEASE_GCC49/AARCH64/MdePkg/Library/CompilerIntrinsicsLib/CompilerIntrinsicsLib]


build.py...
 : error F002: Failed to build module
	/home/richard/Workspace/sct_workspace/edk2/MdePkg/Library/StackCheckLibNull/StackCheckLibNull.inf [AARCH64, GCC49, RELEASE]
```

**Comment by os-d** [comment]
@r1chard-lyu can you share your build log? Also GCC49 is very old and deprecated from edk2. You should be using GCC5 or GCC. I don't know if edk2-test needs to be updated for that.

**Comment by r1chard-lyu** [comment]
@os-d Yes, the attachment is my [build.log](https://github.com/user-attachments/files/18016452/build.log). By the way, I'm not sure why the command `SctPkg/build.sh AARCH64 GCC RELEASE` specifies GCC49, even though my GCC version is 12.3.0. I also tried running `SctPkg/build.sh AARCH64 GCC RELEASE` again, but it still uses GCC49 to build. Could this be an issue with the GCC version being too old?

**Reply by os-d** [reply, marked as answer]
@r1chard-lyu , this looks like a toolchain misconfigured issue, since it seems that it isn’t understanding “make”. GCC49 is simply the set of definitions in tools_def.template that is being used, not which specific GCC compiler. It is worth trying GCC or GCC5 as TOOL_CHAIN_TAG because it does use different environment variables, but I would make sure your GCC_PREFIX env variables are pointing to the right GCC files, i.e. is there anything there.

**Reply by r1chard-lyu** [reply]
@os-d Thank you for your suggestion, I have already solved this problem!
